pkg/telemetry/connection_test.go

```diff
@@ -0,0 +1,89 @@
+package telemetry_test
+
+import (
+	"errors"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	message "phantom-fleet/pkg/msg"
+	"phantom-fleet/pkg/telemetry"
+)
+
+// Mock WebSocketConn for testing
+type mockWebSocketConn struct {
+	WriteMessageFn func(messageType int, data []byte) error
+	CloseFn        func() error
+}
+
+func (m *mockWebSocketConn) WriteMessage(messageType int, data []byte) error {
+	return m.WriteMessageFn(messageType, data)
+}
+
+func (m *mockWebSocketConn) Close() error {
+	return m.CloseFn()
+}
```

Comment:
Mock connections go in /test/mocks

Reply:
Also, use "github.com/stretchr/testify/mock"

See: [test/mocks/connection.go](test/mocks/connection.go)